nixos/modules/virtualisation/amazon-init.nix

```diff
@@ -41,20 +45,45 @@ let
     nixos-rebuild switch
   '';
 in {
-  systemd.services.amazon-init = {
-    inherit script;
-    description = "Reconfigure the system from EC2 userdata on startup";
 
-    wantedBy = [ "multi-user.target" ];
-    after = [ "multi-user.target" ];
-    requires = [ "network-online.target" ];
+  options.virtualisation.amazon-init = {
+    enable = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Enable or disable the amazon-init service.
+      '';
+    };
+
+    script = mkOption {
+      default = script;
+      type = types.str;
```

Comment:
I am really unsure about this type. Is it lines that get executed or a file? Maybe adding an example would be helpful, too.

Reply:
This is a minimal change compared to before. The text of the script is stored in the `script` variable. The default is current behavior.

Reply:
@SuperSandro2000: You're right, the actual type is `lines`.

To dive deeper, this option is used in `systemd.services.amazon-init.script`, which is a string option defined in `nixos/modules/system/boot/systemd.nix` (and that is of type `lines`.

I changed this now and also added an example.

PTAL